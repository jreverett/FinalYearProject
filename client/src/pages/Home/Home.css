#home-container {
  height: 60vh;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

#home-search-container {
  display: flex;
  width: 24rem;
  flex-direction: column;
}

#home-upper-text {
  font-size: 150%;
  margin-bottom: 0;
}

#home-event-count {
  align-self: flex-end;
}
