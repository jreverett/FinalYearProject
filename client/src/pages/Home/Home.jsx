import React, { Component } from 'react';
import { EventSearch } from '../../components';
import { eventService } from '../../services';
import '../../common.css';
import './Home.css';

class Home extends Component {
  constructor(props) {
    super(props);

    this.state = {
      numEvents: '',
    };
  }

  componentDidMount() {
    eventService.getEventCount().then((count) => {
      if (Number.isInteger(count.data))
        this.setState({ numEvents: count.data });
    });
  }

  render() {
    const { numEvents } = this.state;
    return (
      <div id="home-container">
        <div id="home-search-container">
          <p id="home-upper-text">Discover an amazing...</p>
          <EventSearch updateSearchTitle={this.props.updateSearchTitle} />
          {numEvents && (
            <p id="home-event-count">{numEvents} events and counting</p>
          )}
        </div>
      </div>
    );
  }
}

export default Home;
