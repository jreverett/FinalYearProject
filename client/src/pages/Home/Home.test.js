import React from 'react';
import renderer from 'react-test-renderer';
import Home from './Home';

describe('<Home />', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
