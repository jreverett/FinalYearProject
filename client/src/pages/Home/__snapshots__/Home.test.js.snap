// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> matches the snapshot 1`] = `
<div
  id="home-container"
>
  <div
    id="home-search-container"
  >
    <p
      id="home-upper-text"
    >
      Discover an amazing...
    </p>
    <form
      id="event-search-container"
      onSubmit={[Function]}
    >
      <label
        id="event-search-label"
      >
        <input
          className="event-search-text"
          id="event-search-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          value=""
        />
        <span
          className="event-search-text"
          id="event-search-input-overlay"
        >
          <div
            className="Typist "
          >
            <span
              className="Cursor Cursor--blinking"
            >
              |
            </span>
          </div>
        </span>
      </label>
    </form>
    
  </div>
</div>
`;
